### PR TITLE
feat(payment): INT-5467 Stripe UPE GooglePay

### PR DESCRIPTION
**Pull request by BC-PabloRivera**
## What? [INT-5467](https://jira.bigcommerce.com/browse/INT-5467)
Created strategies for GooglePay with StripeUPE

## Why?
To be able to pay using GooglePay via Stripe UPE

## Testing / Proof
image
image
image
image

@bigcommerce/checkout @bigcommerce/payments

### CODE REVIEW

**Review by JoseAlvarezHdz — APPROVED** [review]
LGTM 👍🏾

File: src/checkout-buttons/strategies/googlepay/googlepay-button-strategy.ts [hunk 1]
```diff
@@ -100,6 +100,10 @@ export default class GooglePayButtonStrategy implements CheckoutButtonStrategy {
             return options.googlepaystripe;
         }
 
+        if (options.methodId === 'googlepaystripeupe' && options.googlepaystripeupe) {
```
Comment (animesh1987) [inline src/checkout-buttons/strategies/googlepay/googlepay-button-strategy.ts:103]: Why not use the method defined over here https://github.com/bigcommerce/checkout-sdk-js/pull/1333/files#diff-1c030e40842d05c5b7e7d0f4727ca2e3394c61401da70a728c0db2665bb597b3R12?

File: src/customer/strategies/googlepay/googlepay-customer-strategy.ts [hunk 2]
```diff
@@ -115,6 +115,10 @@ export default class GooglePayCustomerStrategy implements CustomerStrategy {
             return options.googlepaystripe;
         }
 
+        if (options.methodId === 'googlepaystripeupe' && options.googlepaystripeupe) {
```
Comment (animesh1987) [inline src/customer/strategies/googlepay/googlepay-customer-strategy.ts:118]: Same as above

File: src/payment/strategies/googlepay/googlepay-stripe-upe-initializer.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+    teardown(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    parseResponse(paymentData: GooglePaymentData): Promise<TokenizePayload> {
+        try {
+            const payload = JSON.parse(paymentData.paymentMethodData.tokenizationData.token);
+
+            return Promise.resolve({
```
Comment (animesh1987) [inline src/payment/strategies/googlepay/googlepay-stripe-upe-initializer.ts:30]: Can we have some typeguard for payload to check if it actually has `id` and `type` values?

Same for `card` below?

File: src/payment/strategies/googlepay/googlepay-stripe-upe-initializer.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                        gateway: 'stripe',
+                        'stripe:version': stripeVersion,
+                        'stripe:publishableKey': `${stripePublishableKey}/${stripeConnectedAccount}`,
+                    },
+                },
+            }],
+            transactionInfo: {
+                currencyCode,
+                totalPriceStatus: 'FINAL',
+                totalPrice: round(outstandingBalance, 2).toFixed(2),
```
Comment (animesh1987) [inline src/payment/strategies/googlepay/googlepay-stripe-upe-initializer.ts:98]: Here to round we should use the decimal precision for the currency.

Comment (BC-PabloRivera) [reply src/payment/strategies/googlepay/googlepay-stripe-upe-initializer.ts]: We are using outstandingBalance because thats the grandTotal - store credit

**Comment by BC-PabloRivera** [comment]
♻️  @animesh1987 please check out the changes I made, is that what you had in mind?

**Review by animesh1987 — APPROVED** [review]
Nice work

**Pull request merged**